tests/csapi/apidoc_content_test.go

```diff
@@ -27,3 +27,23 @@ func TestContent(t *testing.T) {
 		t.Fatalf("expected contentType to be %s, got %s", wantContentType, contentType)
 	}
 }
+
+// same as above but testing _matrix/client/v1/media/download
+func TestContentCSAPIMediaV1(t *testing.T) {
+	deployment := complement.Deploy(t, 2)
+	defer deployment.Destroy(t)
+
+	hs1 := deployment.Register(t, "hs1", helpers.RegistrationOpts{})
+	hs2 := deployment.Register(t, "hs2", helpers.RegistrationOpts{})
+
+	wantContentType := "img/png"
+	mxcUri := hs1.UploadContent(t, data.MatrixPng, "test.png", wantContentType)
+
+	content, contentType := hs2.DownloadContentAuthenticated(t, mxcUri)
+	if !bytes.Equal(data.MatrixPng, content) {
+		t.Fatalf("uploaded and downloaded content doesn't match: want %v\ngot\n%v", data.MatrixPng, content)
+	}
+	if contentType != wantContentType {
+		t.Fatalf("expected contentType to be \n %s, got \n %s", wantContentType, contentType)
+	}
```

Comment:
Oh, I just saw that this is testing the federated endpoints as well? (upload on "hs1", download from "hs2")

We probably want to only check from "hs1" here, as this is in the "csapi" directory.

Also, given this endpoint requires authentication now, we should remove the access token from the user and test that we receive an unauthorized response.
i.e.:

```go
// same as above but testing _matrix/client/v1/media/download
func TestContentCSAPIMediaV1(t *testing.T) {
	deployment := complement.Deploy(t, 1)
	defer deployment.Destroy(t)

	alice := deployment.Register(t, "hs1", helpers.RegistrationOpts{})

	wantContentType := "img/png"
	mxcUri := alice.UploadContent(t, data.MatrixPng, "test.png", wantContentType)

	status, content, contentType := alice.DownloadContentAuthenticated(t, mxcUri)
	if status != http.StatusOK {
		t.Fatalf("expected HTTP status: %d, got %d", http.StatusOK, status)
	}
	if !bytes.Equal(data.MatrixPng, content) {
		t.Fatalf("uploaded and downloaded content doesn't match: want %v\ngot\n%v", data.MatrixPng, content)
	}
	if contentType != wantContentType {
		t.Fatalf("expected contentType to be \n %s, got \n %s", wantContentType, contentType)
	}

	// Remove the AccessToken and try again, this should now return a 401.
	alice.AccessToken = ""
	status, _, _ = alice.DownloadContentAuthenticated(t, mxcUri)
	if status != http.StatusUnauthorized {
		t.Fatalf("expected HTTP status: %d, got %d", http.StatusUnauthorized, status)
	}
}
```

`DownloadContentAuthenticated` needs to be updated to use `c.Do` instead of `c.MustDo`. `res.StatusCode` can then be returned as the first return value.

Reply:
I didn't re-write `DownloadContentAuthenticated` but just used `c.Do` directly to test the response code when the access token is missing - it seemed clearer to me.

Reply:
Oh, yea, that makes more sense.